core/src/providers/anthropic/anthropic.rs

```diff
@@ -326,6 +329,10 @@ impl LLM for AnthropicLLM {
         }
     }
 
+    fn set_tokenizer_from_config(&mut self, config: crate::types::tokenizer::TokenizerConfig) {
+        self.tokenizer = TokenizerSingleton::from_config(config.clone());
```

Comment:
Do you actually need to clone ? My guess is that `TokenizerSingleton::from_config` could be updated to just take a ref (`&Thing`) instead of a full config which would remove the need to clone.